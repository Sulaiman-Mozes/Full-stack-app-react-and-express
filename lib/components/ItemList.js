import React from 'react';
import PropTypes from 'prop-types';
import Article from './Article';

const ItemList = (props) => {
  const { articles } = props;
  return (
    <div>
      {Object.values(articles).map(article => (
        <Article
          key={article.id}
          article={article}
        />
      ))}
    </div>
  );
};

ItemList.propTypes = {
  articles: PropTypes.shape({}).isRequired,
};

export default ItemList;
