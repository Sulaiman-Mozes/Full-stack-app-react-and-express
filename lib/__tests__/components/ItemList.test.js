import React from 'react';
import { shallow } from 'enzyme';
import ItemList from '../../components/ItemList';


describe('Render Component', () => {
  it('It should render ItemList Component', () => {
    const props = {
      articles: {
        1: {
          id: 1, title: 'title', date: '', body: 'body',
        },
      },
      store: {
        lookupAuthor: jest.fn,
      },
    };
    const wrapper = shallow(<ItemList {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
