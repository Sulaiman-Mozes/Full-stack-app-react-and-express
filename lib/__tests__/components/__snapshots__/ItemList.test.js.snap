// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Component It should render ItemList Component 1`] = `
<div>
  <props => {
  /* istanbul ignore next */
  cov_1n8pcmooa0.f[1]++;
  const {
    article,
    store: {
      lookupAuthor
    }
  } =
  /* istanbul ignore next */
  (cov_1n8pcmooa0.s[4]++, props);
  const author =
  /* istanbul ignore next */
  (cov_1n8pcmooa0.s[5]++, lookupAuthor(article.authorId));
  const {
    lastName,
    firstName,
    website
  } =
  /* istanbul ignore next */
  (cov_1n8pcmooa0.s[6]++, author);

  /* istanbul ignore next */
  cov_1n8pcmooa0.s[7]++;
  return (
    /* istanbul ignore next */
    _react.default.createElement(
    /* istanbul ignore next */
    "div",
    /* istanbul ignore next */
    {
      style: styles.article
    },
    /* istanbul ignore next */
    _react.default.createElement(
    /* istanbul ignore next */
    "div",
    /* istanbul ignore next */
    {
      style: styles.title
    }, article.title),
    /* istanbul ignore next */
    _react.default.createElement(
    /* istanbul ignore next */
    "div",
    /* istanbul ignore next */
    {
      style: styles.date
    }, dateDisplay(article.date)),
    /* istanbul ignore next */
    _react.default.createElement(
    /* istanbul ignore next */
    "div",
    /* istanbul ignore next */
    {
      style: styles.author
    },
    /* istanbul ignore next */
    _react.default.createElement(
    /* istanbul ignore next */
    "a",
    /* istanbul ignore next */
    {
      href: website
    }, \`\${firstName} \${lastName}\`)),
    /* istanbul ignore next */
    _react.default.createElement(
    /* istanbul ignore next */
    "div",
    /* istanbul ignore next */
    {
      style: styles.body
    }, article.body))
  );
}Container
    article={
      Object {
        "body": "body",
        "date": "",
        "id": 1,
        "title": "title",
      }
    }
    key="1"
  />
</div>
`;
